config/viper_source.go

```diff
@@ -0,0 +1,36 @@
+package config
+
+import "github.com/spf13/viper"
+
+// ViperSource implements the Source interface using the viper package for
+// configuration file.
+type ViperSource struct {
+	v *viper.Viper
+}
+
+// NewViperSource returns a new ViperSource from the top level Viper object,i.e.
+// it calls viper.GetViper(). The caller must configure the viper package to
+// give the path and names of the config files to search for. It can be done
+// with:
+//
+//    viper.SetConfig("name")
+//    viper.SetConfigPath(".")
+//
+func NewViperSource() *ViperSource {
+	return &ViperSource{viper.GetViper()}
+}
+
+// Defined returns true if the key is defined in the configuration file
+func (v *ViperSource) Defined(key string) bool {
+	return v.v.IsSet(key)
```

Comment:
It's a taste issue at this point whether we prefer `IsSet` or `Defined` because I'll anyway need to define `ViperSource` since I don't want the `GetString` but just the `String` method.